src/Analysis/GWASResults/Views/Results/ResultsPheWeb/TopLociTable/TopLociTable.jsx

```diff
@@ -0,0 +1,187 @@
+import React, { useMemo, useState, useEffect } from 'react';
+import { Table, Input, Button } from 'antd';
+import PropTypes from 'prop-types';
+import { SearchOutlined } from '@ant-design/icons';
+import downloadTSVFromJson from './downloadTSVFromJSON';
+import filterLociTableData from './filterLociTableData';
+
+const TopLociTable = ({ data }) => {
+  // Adds a variant key value pair with the desired formatting
+  const tableData = useMemo(
+    () => data.map((obj) => ({
+      ...obj,
+      variant: `${obj?.chrom}:${obj?.pos.toLocaleString('en-US')} ${obj?.ref}/
+    ${obj?.alt} (${obj?.rsids})`,
+    })),
+    [data],
+  );
+
+  const [filteredData, setFilteredData] = useState(data);
+  const [lociTableState, setLociTableState] = useState({
+    variantSearchTerm: '',
+    nearestGenesSearchTerm: '',
+    pvalSearchTerm: '',
+    afSearchTerm: '',
+    currentPage: 1,
+  });
+
+  useEffect(() => {
+    setFilteredData(filterLociTableData(tableData, lociTableState));
+  }, [lociTableState, tableData]);
+
+  const handleSearchTermChange = (event, searchTermKey) => {
+    if (searchTermKey === 'variant') {
+      setLociTableState({
+        ...lociTableState,
+        currentPage: 1,
+        variantSearchTerm: event.target.value,
+      });
+    }
+    if (searchTermKey === 'nearest_genes') {
+      setLociTableState({
+        ...lociTableState,
+        currentPage: 1,
+        nearestGenesSearchTerm: event.target.value,
+      });
+    }
+    if (searchTermKey === 'pval') {
+      setLociTableState({
+        ...lociTableState,
+        currentPage: 1,
+        pvalSearchTerm: event.target.value,
+      });
+    }
+    if (searchTermKey === 'af') {
+      setLociTableState({
+        ...lociTableState,
+        currentPage: 1,
+        afSearchTerm: event.target.value,
+      });
+    }
+    return null;
+  };
+
+  const handleTableChange = (pagination) => {
+    if (pagination.current !== lociTableState.currentPage) {
+      // User changes page selection, set page to current pagination selection
+      return setLociTableState({
+        ...lociTableState,
+        currentPage: pagination.current,
+      });
+    }
+    // When the user updates sorting set page to first page
+    return setLociTableState({
+      ...lociTableState,
+      currentPage: 1,
+    });
+  };
+
+  const columns = [
+    {
+      title: 'Variant',
+      dataIndex: 'variant',
+      key: 'variant',
+      sorter: (a, b) => a.variant.localeCompare(b.variant),
```

Comment:
we'll need a separate/custom sorter for this. Now we get the chromosomes sorted like this:
![Screenshot 2023-06-21 at 19 33 25](https://github.com/uc-cdis/data-portal/assets/2900303/45bdeba8-56a9-407b-925d-aacd1010d649)

1, 11, .. 2, 20, ...3, etc

Reply:
Similar problem in the p-value column: 

![Screenshot 2023-06-21 at 19 44 41](https://github.com/uc-cdis/data-portal/assets/2900303/b8dea592-d5da-4a5d-a142-94a5d22091ce)

Reply:
The sorting algorithms have been updated. 

For the PVAL and AF the number is parsed as a number before doing the comparison, which should make numbers in scientific notation render in the correct order. 

For example: 
```
  sorter: (a, b) => Number(a.pval) - Number(b.pval),
  ```


For the variant column, the numbers are sorted first by the chrom value, and if those match, then the sorting is followed by the pos value. 
For example: 
```
      sorter: (a, b) => {
        const chromA = Number(a.variant.split(':')[0]);
        const chromB = Number(b.variant.split(':')[0]);
        if (chromA === chromB) {
          let posA = (a.variant.split(':')[1]);
          posA = posA.substring(0, posA.indexOf(' '));
          posA = Number(posA.replace(/,/g, ''));

          let posB = (b.variant.split(':')[1]);
          posB = posB.substring(0, posB.indexOf(' '));
          posB = Number(posB.replace(/,/g, ''));
          return posA - posB;
        }
        return chromA - chromB;
      },
      ```